File: swift/Sources/benchmarks/Matmul16T.swift
import Dispatch
import Foundation

final class Matmul16T: Matmul4T {
  override init() {
    super.init()
    n = configValue("n") ?? 0
  }

  override func getNumThreads() -> Int {
    return 16
  }

  override func name() -> String {
    return "Matmul::T16"
  }
}

File: java/src/main/java/benchmarks/Matmul16T.java
package benchmarks;

import java.util.concurrent.ForkJoinPool;
import java.util.stream.IntStream;

public class Matmul16T extends Matmul4T {

    public Matmul16T() {
        n = (int) configVal("n");
        resultVal = 0L;
        POOL = new ForkJoinPool(16);
    }

    @Override
    public String name() {
        return "Matmul::T16";
    }
}